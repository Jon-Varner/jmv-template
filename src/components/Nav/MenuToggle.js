import React from 'react';
import PropTypes from 'prop-types';

const menuToggle = ({ menuOpen, toggleMenu }) => (
  <button
    className="menu-toggle"
    aria-label="Open the menu"
    onClick={toggleMenu}
  >
    <span className="menu-hamburger" aria-hidden="true" />
    <span className="menu-hamburger" aria-hidden="true" />
    <span className="menu-hamburger" aria-hidden="true" />
  </button>
);

menuToggle.propTypes = {
  menuOpen: PropTypes.bool,
  toggleMenu: PropTypes.func
};

export default menuToggle;
