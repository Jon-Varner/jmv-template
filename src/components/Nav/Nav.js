import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

import MenuToggle from './MenuToggle';

const nav = ({ menuOpen, toggleMenu }) => (
  <nav className={menuOpen ? 'primary-nav is-open' : 'primary-nav'}>
    <MenuToggle menuOpen={menuOpen} toggleMenu={toggleMenu} />
    <ul>
      <li>
        <NavLink exact to="/">
          Home
        </NavLink>
      </li>
      <li>
        <NavLink exact to="/sample-form">
          Sample Form
        </NavLink>
      </li>
      <li>
        <NavLink exact to="/sample-api-call">
          Sample API Call
        </NavLink>
      </li>
    </ul>
  </nav>
);

nav.propTypes = {
  menuOpen: PropTypes.bool,
  toggleMenu: PropTypes.func
};

export default nav;
