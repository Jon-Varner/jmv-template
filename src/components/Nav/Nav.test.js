import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render, cleanup } from 'react-testing-library';
import 'jest-dom/extend-expect';

import Nav from './Nav';

afterEach(cleanup);

const toggleMenu = jest.fn();

it('renders', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <Nav menuOpen={false} toggleMenu={toggleMenu} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it('defaults to closed menu', () => {
  const { container } = render(
    <MemoryRouter>
      <Nav menuOpen={false} toggleMenu={toggleMenu} />
    </MemoryRouter>
  );

  const menu = container.querySelector('nav');

  expect(menu.classList.contains('is-open')).toBe(false);
});

it('shows menu if prop value is true', () => {
  const { container } = render(
    <MemoryRouter>
      <Nav menuOpen={true} toggleMenu={toggleMenu} />
    </MemoryRouter>
  );

  const menu = container.querySelector('nav');

  expect(menu.classList.contains('is-open')).toBe(true);
});
