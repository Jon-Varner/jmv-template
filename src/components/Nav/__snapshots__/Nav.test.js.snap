// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <nav
    class="primary-nav"
  >
    <button
      aria-label="Open the menu"
      class="menu-toggle"
    >
      <span
        aria-hidden="true"
        class="menu-hamburger"
      />
      <span
        aria-hidden="true"
        class="menu-hamburger"
      />
      <span
        aria-hidden="true"
        class="menu-hamburger"
      />
    </button>
    <ul>
      <li>
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/sample-form"
        >
          Sample Form
        </a>
      </li>
      <li>
        <a
          href="/sample-api-call"
        >
          Sample API Call
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
